moveit_ros/move_group/src/default_capabilities/move_action_capability.cpp

```diff
@@ -54,151 +57,177 @@ MoveGroupMoveAction::MoveGroupMoveAction()
 }
 
-void MoveGroupMoveAction::executeMoveCallback(const moveit_msgs::action::MoveGroupGoalConstPtr& goal)
+void MoveGroupMoveAction::executeMoveCallback(std::shared_ptr<MGActionGoal> goal)
 {
-  setMoveState(PLANNING);
+  RCLCPP_INFO(LOGGER, "executing..");
+  setMoveState(PLANNING, goal);
   // before we start planning, ensure that we have the latest robot state received...
-  context_->planning_scene_monitor_->waitForCurrentRobotState(ros::Time::now());
+  context_->planning_scene_monitor_->waitForCurrentRobotState(rclcpp::Clock().now());
   context_->planning_scene_monitor_->updateFrameTransforms();
 
-  moveit_msgs::action::MoveGroupResult action_res;
-  if (goal->planning_options.plan_only || !context_->allow_trajectory_execution_)
+  auto action_res = std::make_shared<MGAction::Result>();
+  if (goal->get_goal()->planning_options.plan_only || !context_->allow_trajectory_execution_)
   {
-    if (!goal->planning_options.plan_only)
-      ROS_WARN_NAMED(getName(), "This instance of MoveGroup is not allowed to execute trajectories "
-                                "but the goal request has plan_only set to false. "
-                                "Only a motion plan will be computed anyway.");
+    if (!goal->get_goal()->planning_options.plan_only)
+      RCLCPP_WARN(LOGGER, "This instance of MoveGroup is not allowed to execute trajectories "
+                          "but the goal request has plan_only set to false. "
+                          "Only a motion plan will be computed anyway.");
     executeMoveCallbackPlanOnly(goal, action_res);
   }
   else
     executeMoveCallbackPlanAndExecute(goal, action_res);
 
-  bool planned_trajectory_empty = trajectory_processing::isTrajectoryEmpty(action_res.planned_trajectory);
-  std::string response =
-      getActionResultString(action_res.error_code, planned_trajectory_empty, goal->planning_options.plan_only);
-  if (action_res.error_code.val == moveit_msgs::msg::MoveItErrorCodes::SUCCESS)
-    move_action_server_->setSucceeded(action_res, response);
+  bool planned_trajectory_empty = trajectory_processing::isTrajectoryEmpty(action_res->planned_trajectory);
+  // @todo: Response messages
+  std::string response = getActionResultString(action_res->error_code, planned_trajectory_empty,
+                                               goal->get_goal()->planning_options.plan_only);
+  auto fb = std::make_shared<MGAction::Feedback>();
+  fb->state = response;
+  if (action_res->error_code.val == moveit_msgs::msg::MoveItErrorCodes::SUCCESS)
+  {
+    goal->succeed(action_res);
+    goal->publish_feedback(fb);
+  }
   else
   {
-    if (action_res.error_code.val == moveit_msgs::msg::MoveItErrorCodes::PREEMPTED)
-      move_action_server_->setPreempted(action_res, response);
+    if (action_res->error_code.val == moveit_msgs::msg::MoveItErrorCodes::PREEMPTED)
+    {
+      //@todo: wait for preempt?
+      // move_action_server_->setPreempted(action_res, response);
```

Comment:
Shouldn't the goal just get canceled here?

Reply:
iirc in ros1 there was setCanceled and setPreempted, i wasn't sure what the difference was. anyway, i changed this to canceled 44c1643